prow/flagutil/github.go

```diff
@@ -114,30 +119,48 @@ func (o *GitHubOptions) GitHubClient(secretAgent *secret.Agent, dryRun bool) (cl
 
 // GitClient returns a Git client.
 func (o *GitHubOptions) GitClient(secretAgent *secret.Agent, dryRun bool) (client *git.Client, err error) {
-	client, err = git.NewClient()
+	old, _, err := o.GitClients(secretAgent, dryRun)
+	return old, err
+}
+
+// GitClients returns both old and new Git clients.
+func (o *GitHubOptions) GitClients(secretAgent *secret.Agent, dryRun bool) (old *git.Client, new git.Clients, err error) {
+	old, err = git.NewClient()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	// We must capture the value of client here to prevent issues related
 	// to the use of named return values when an error is encountered.
 	// Without this, we risk a nil pointer dereference.
 	defer func(client *git.Client) {
 		if err != nil {
-			client.Clean()
+			if err := client.Clean(); err != nil {
+				logrus.WithError(err).Warn("Failed to clean up git client.")
+			}
 		}
-	}(client)
+	}(old)
 
 	// Get the bot's name in order to set credentials for the Git client.
 	githubClient, err := o.GitHubClient(secretAgent, dryRun)
 	if err != nil {
-		return nil, fmt.Errorf("error getting GitHub client: %v", err)
+		return nil, nil, fmt.Errorf("error getting GitHub client: %v", err)
 	}
 	botName, err := githubClient.BotName()
 	if err != nil {
-		return nil, fmt.Errorf("error getting bot name: %v", err)
+		return nil, nil, fmt.Errorf("error getting bot name: %v", err)
 	}
-	client.SetCredentials(botName, secretAgent.GetTokenGenerator(o.TokenPath))
+	old.SetCredentials(botName, secretAgent.GetTokenGenerator(o.TokenPath))
 
-	return client, nil
+	if o.gitHost == "" {
+		return nil, nil, errors.New("git client request but --git-host not provided")
+	}
+	adapter := func() (name, email, login string, err error) {
+		user, err := githubClient.BotUser()
+		if err != nil {
+			return "", "", "", err
+		}
+		return user.Name, user.Email, user.Login, nil
```

Comment:
This isn't sufficient, we need to allow at least the email to be explicitly specified since that is not always exposed over the `/user` API endpoint (name may be the same).